### ISSUE TITLE
[a11y]: SidePanel component missing tabindex="0" when scrollable

### ISSUE THREAD

**Issue by David-Yuen** [carbon-design-system/ibm-products#7043]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.59.0

### React version

18.3.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The `SidePanel` has an accessibility error `"The scrollable element <div> with non-interactive content is not tabbable"` when the scroll bar appears on the body of the side panel body.
Specifically, the side panel needs to have read-only content. If there is interactive components (like input or button or something), you would be able to tab to it.. But this error is specific when there is only read-only content, nothing interactive. 

Using the reproduction example as reference, i purposely only put in a few read-only content on the side panel. 
If you load this page, open the side panel, and generally with a normal sized browser, all content will fit on page and there is no scrollbar. 
- At this point, you can only tab to the `close` button
- And if you run the scan now, it is clean
<img width="1235" alt="Image" src="https://github.com/user-attachments/assets/418a0f42-2248-4289-912e-2f0be7482add" />

Now shrink the browser height so it is smaller than the content in the side panel.. this is to enable the scrollbar, like this:
<img width="1237" alt="Image" src="https://github.com/user-attachments/assets/f02d11dd-63a6-4766-96f6-a62bd6eeb9dd" />

Now that the scrollbar appears, you can now also tab to the side panel's body section, and use the keyboard up/down arrow keys to scroll up and down. As shown here, i tabbed to the side panel body section and it has a blue border now. 
<img width="535" alt="Image" src="https://github.com/user-attachments/assets/f7979d9b-5918-4153-bc7f-8dae76dc6fac" />

This is good, but if you run the scan now, it finds an error `The scrollable element <div> with non-interactive content is not tabbable`. 

I've discussed with the accessibility team, and all that's needed to resolve this is add `tabindex="0"` to the `div` with `className="c4p--side-panel__inner-content c4p--side-panel--scrolls"`..  Like this:
<img width="491" alt="Image" src="https://github.com/user-attachments/assets/f5950f79-a93e-4317-be02-d29142f0e0b8" />

Note that `tabindex` should be set dynamically.. where if the scrollbar is visible then use `tabindex="0"`. And if the scrollbar is not visible, then use `tabindex="-1"`.

With the `tabindex` added, the scan is clean

### Reproduction/example

https://stackblitz.com/edit/github-3qnrvnl8?file=src%2FApp.jsx,src%2Findex.scss

### Steps to reproduce

Load the reproduction example
Click Open side panel button to open side panel
While side panel is open, shrink the window height smaller than the content in the side panel. This is to enable the scrollbar of the side panel body.. Like this:
<img width="1234" alt="Image" src="https://github.com/user-attachments/assets/93935779-c521-4bdd-8203-2aeb87cb00ea" />

While the side panel and it's scrollbar is visible, run the scan. You will get a scan error:
<img width="547" alt="Image" src="https://github.com/user-attachments/assets/b80903bc-c306-470c-8f9d-f709616d56cb" />

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by jlongshore** [comment]
Hey David, thanks for the in-depth report!

**Note to devs:** reference AboutModal for similar implementation.

**Comment by wkeese** [comment]
> "The scrollable element <div> with non-interactive content is not tabbable"
> ..
> I've discussed with the accessibility team, and all that's needed to resolve this is add tabindex="0" to the div with className="c4p--side-panel__inner-content c4p--side-panel--scrolls".. 

Not sure exactly what your conversation was, but I think they meant to only add tabindex=0 **when there's no interactive content**.

Notice how that's mentioned in the error message (and in related documentation from WCAG etc.).

@carbon/react's TabPanel tries to be clever and monitor the content using `MutationObserver`, checking if there are interactive controls, and correspondingly switching between tabindex=0 and tabindex=-1.  It doesn't seem to be currently working though.

The simpler solution would be to just let the application specify the tabindex.

**Comment by David-Yuen** [comment]
Currently there is no option to pass-in or set the `tabIndex` from the application. If `SidePanel` added a prop that we can pass in to set `tabIndex=0` or `tabIndex=-1`, then we could figure out when it's needed or not.
Some instances of the SidePanel we only have read-only content (no interactive components.. so nothing to tab to), and some instances we do have some buttons or other interactive content.

**Comment by wkeese** [comment]
Wow, that's a curveball.   I can't see that slack conversation but I can guess what it says.  In that case, let's simplify the Carbon code.   I started with https://github.com/carbon-design-system/carbon/pull/19405.

**Comment by wkeese** [comment]
I filed https://github.com/IBMa/equal-access/issues/2281 to hopefully have that check removed from accessiblity-checker, although there's some debate about it since Mac Safari still doesn't support it.
